File: Programmers/Lv.1/비밀지도/17681.java
class Solution {
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        for (int i=0; i<n; i++) {
            String[] arr = new String[n];
            
            for (int j=n-1; j>=0; j--) {
                int num = (int) Math.pow(2, j);
                if (arr1[i] >= num) {
                    arr[n - j - 1] = "#";
                    arr1[i] -= num;
                }
                
                if (arr2[i] >= num) {
                    arr[n - j - 1] = "#";
                    arr2[i] -= num;
                }
            }
            
            answer[i] = String.join("", arr).replaceAll("null", " ");
        }
        
        return answer;
    }
}

File: Programmers/Lv.1/비밀지도/17681.js
function solution(n, arr1, arr2) {
    const answer = [];
    
    for (let i=0; i<n; i++) {
        const arr = new Array(n).fill(" ");
        
        for (let j=n-1; j>=0; j--) {
            const num = Math.pow(2, j);
            if (arr1[i] >= num) {
                arr[n - j - 1] = "#";
                arr1[i] -= num;
            }
            
            if (arr2[i] >= num) {
                arr[n - j - 1] = "#";
                arr2[i] -= num;
            }
        }
        
        answer.push(arr.join(""));
    }
    
    return answer;
}
